torch/csrc/jit/script/compiler.cpp

```diff
@@ -1314,12 +1314,15 @@ struct to_ir {
         Value* cur_elem = iter_val->getelem(range, method, trip_count);
         SugaredValuePtr sv = std::make_shared<SimpleValue>(cur_elem);
         List<Expr> target_exprs = targets.value();
-        size_t n_binders = target_exprs.size();
+        validateAssignLhsExpr(target_exprs, range);
 
-        bool starred_unpack = calcNumStarredUnpack(target_exprs, range);
-        if (starred_unpack)
-          n_binders--;
-        emitExprsAssign(target_exprs, {sv}, range, n_binders);
+        // if target exprs are more than 1, it means iteration unpacking on LHS
+        // we create Tuple literal to wrap those target exprs for assignments
+        if (target_exprs.size() > 1) {
```

Comment:
Why is it necessary to create a list here vs just passing more than one target_expr to emitExprsAssign?

Reply:
Oh it's because `sv` here is the root IterableTree (which create TupleConstruct in `getelem`) and need to call emitTupleAssign to unwrap the tuple using `asTuple`, but the target_exprs here is not a tupleliteral, so `emitExprsAssign` could not match these two together, and we need to create a TupleLiteral wrap here.